### PR TITLE
Slack auth

### PR DESCRIPTION
**Pull request by batuncer**
Hello Guys,

I created login page and sigh up page.
Slack sign up client side almost done. I will share it which documents i read and which code. 
I import some file during slack issue. Those are already in this code. You can try and if it doesnt work we can check them on saturday. 

You can see loginpage sending /login parameter.

Thanks

### CODE REVIEW

File: Back-End-Node/package-lock.json
Comment (AnnaFYZ) [inline Back-End-Node/package-lock.json:1]: you are right, this one should be change, local host is not going to work anyway

File: Back-End-Node/server.js [hunk 1]
```diff
@@ -1,23 +1,126 @@
 const express = require("express");
 const app = express();
 const { pool } = require("./dbConfig");
-
+var http = require("http");
```
Comment (AnnaFYZ) [inline Back-End-Node/server.js:4]: we need to change all "var"s with const or let, this really old version of declaring a variable and is not used in JavaScript any more, but i think you took it from C# , did not you :wink:

File: Back-End-Node/server.js [hunk 2]
```diff
@@ -1,23 +1,126 @@
...
+        headers: {
+          Authorization: `Bearer ${authJson.authed_user.access_token}`,
+        },
+      }
+    );
+    const userDataJson = await userDataResponse.json();
+   
+    console.log("userDataJson", userDataJson);
+
+    res.redirect("http://localhost:3000/oauthdone?code=1234");
```
Comment (AnnaFYZ) [inline Back-End-Node/server.js:53]: will localhost work here?

File: front-end/src/components/header/header.jsx [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+      <AppBar position="static">
+        <Toolbar>
+          <IconButton
+            size="large"
+            edge="start"
+            color="inherit"
+            aria-label="menu"
+            sx={{ mr: 2 }}
+          >
+            <MenuIcon />
```
Comment (AnnaFYZ) [inline front-end/src/components/header/header.jsx:18]: why do we have toolbar inside header? I thought it should be a separate component

File: front-end/src/routes/index.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
 // MainRoutes.jsx
 import React from "react";
 import { Routes, Route } from "react-router-dom";
-import Login from "../pages/login/login.jsx";
-import Header from "../components/header/header.jsx";
+import LoginPage from "../pages/login/login.jsx";
+import SignUpPage from "../pages/signup/signup.jsx";
```
Comment (AnnaFYZ) [inline front-end/src/routes/index.js:5]: sorry, i don't get the point of signup page here. Is not it the same as login page? if you use Slack authentication, all users should be signed up already or is it just a component for future updating events?

**Review by AnnaFYZ — COMMENTED** [review]
WOW!! you did a lot!!! I am looking forward to hear from you how all this works :)

File: Back-End-Node/server.js (hunk 1 above)
Comment (batuncer) [reply Back-End-Node/server.js:4]: It is just copy paste from some another examples. I will change those with next pull request.

File: Back-End-Node/server.js (hunk 2 above)
Comment (batuncer) [reply Back-End-Node/server.js:53]: ofcourse it is not working we need to create that endpoint. It is just an example to replace it with real one.

File: front-end/src/components/header/header.jsx (hunk 3 above)
Comment (batuncer) [reply front-end/src/components/header/header.jsx:18]: We need to speak about design for the first page. I added it from mui material

File: front-end/src/routes/index.js (hunk 4 above)
Comment (batuncer) [reply front-end/src/routes/index.js:5]: I was thinking to add manual sign up too. They are both almost ready. Once we can decide about it if we want to add or not.

File: Back-End-Node/client-cert.pem
Comment (jxz12) [inline Back-End-Node/client-cert.pem:1]: these secret files should not be stored in github, they should be kept secret via render's secret files

**Pull request closed without merging**